src/averageAges.js

```diff
@@ -55,7 +83,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  people.map(function(person) {
+    const parentName = person.name;
+
+    person.child = people.filter(child => child.mother === parentName);
+  });
+
+  const ageDifference = [];
+
+  people.forEach(function(person) {
+    let children = [...person.child];
+
+    if (onlyWithSon) {
+      children = children.filter(child => child.sex !== 'f');
+    }
+
+    const parentBornYear = person.born;
+
+    children.map(function(child) {
+      ageDifference.push(child.born - parentBornYear);
+    });
+  });
```

Comment:
we can search for all children: 
```
const children = people.filter(child => people.some(person =>
    child.mother === person.name
    && (!onlyWithSon || child.sex === 'm'))
  );
```

And then we can find for each one a mother (when mapping for ages )